src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs

```diff
@@ -49,28 +79,27 @@ public static void EnableAadSigningKeyIssuerValidation(this TokenValidationParam
         /// </summary>
         /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
         /// <param name="securityToken">The <see cref="SecurityToken"/> being validated, could be a JwtSecurityToken or JsonWebToken.</param>
-        /// <param name="configuration">The <see cref="OpenIdConnectConfiguration"/> provided.</param>
-        /// <returns><c>true</c> if the issuer signing key is valid; otherwise, <c>false</c>.</returns>
-        internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityToken securityToken, BaseConfiguration configuration)
+        /// <param name="configuration">The <see cref="BaseConfiguration"/> provided.</param>
+        /// <returns><c>true</c> if the issuer of the signing key is valid; otherwise, <c>false</c>.</returns>
+        internal static bool ValidateSigningKeyIssuer(SecurityKey securityKey, SecurityToken securityToken, BaseConfiguration configuration)
         {
             if (securityKey == null)
                 return true;
 
             if (securityToken == null)
                 throw LogHelper.LogArgumentNullException(nameof(securityToken));
 
-            var openIdConnectConfiguration = configuration as OpenIdConnectConfiguration;
-            if (openIdConnectConfiguration == null)
+            if (configuration is not OpenIdConnectConfiguration openIdConnectConfiguration)
                 return true;
 
-            var matchedKeyFromConfig = openIdConnectConfiguration.JsonWebKeySet?.Keys.FirstOrDefault(x => x != null && x.Kid == securityKey.KeyId);
+            JsonWebKey matchedKeyFromConfig = openIdConnectConfiguration.JsonWebKeySet?.Keys.FirstOrDefault(x => x != null && x.Kid == securityKey.KeyId);
```

Comment:
We duplicate this linq code to find the securitykey in ValidateSigningKeyIssuer and ValidateSigningKeyCloudInstanceName. Since this code is on the hot path, we should only find the securitykey once.

We should replace the linq code with simple loops and compares.

Reply:
I changed it a bit, now we don't have single extension method that calls ValidateSigningKeyIssuer and ValidateSigningKeyCloudInstanceName. So, there is nothing to share.

about linq
since the JsonWebKeySet or Keys may be null we have to check for null this properties, so the 'loop' version would not differ from FirstOrDefault.

so, I left FirstOrDefault.
Please have a look and let me know if we can improve it

Reply:
@alexholub113 it has been shown that linq is much slower that simple loops.

Reply:
makes sense, I added new method GetJsonWebKeyBySecurityKey that using loop